deluge/httpdownloader.py

```diff
@@ -52,43 +79,58 @@ def __init__(
         :param headers: any optional headers to send
         :type headers: dictionary
         """
+        if not headers:
+            headers = {}
+        else:
+            for key, value in headers.items():
+                headers[key] = [value]
+        headers.update({'User-Agent': ['Deluge/%s (http://deluge-torrent.org)' % get_version()]})
 
+        self.url = bytes(url)
```

Comment:
use `.encode()` please

Reply:
Do you mean `self.url = bytes(url.encode('utf-8'))`? Or something else?

Reply:
Even simpler `self.url = url.encode()`, default ascii encoding as a url strictly shouldn't have non-ascii chars.